packages/core/src/utilities/VoxelValue.ts

```diff
@@ -0,0 +1,235 @@
+import type { BoundsIJK, Point3, VolumeScalarData } from '../types';
+
+/**
+ * This is a simple, standard interface to values associated with a voxel.
+ */
+export default class VoxelValue<T> {
+  public modifiedSlices = new Set<number>();
+  public boundsIJK = [
+    [Infinity, -Infinity],
+    [Infinity, -Infinity],
+    [Infinity, -Infinity],
+  ] as BoundsIJK;
+
+  // Provide direct access to the underlying data, if any
+  public scalarData: VolumeScalarData;
+  public map: Map<number, T>;
+  public sourceVoxelValue: VoxelValue<T>;
+  public isInObject: (pointIPS, pointIJK) => boolean;
+
+  points: Set<number>;
+  dimensions: Point3;
+  width: number;
+  frameSize: number;
+  _get: (index: number) => T;
+  _set: (index: number, v: T) => boolean | void;
+
+  constructor(
+    dimensions,
+    _get: (index: number) => T,
+    _set?: (index: number, v: T) => boolean | void
+  ) {
+    this.dimensions = dimensions;
+    this.width = dimensions[0];
+    this.frameSize = this.width * dimensions[1];
+    this._get = _get;
+    this._set = _set;
+  }
+
+  public getIJK = (i, j, k) => {
```

Comment:
Lots of these methods should be renamed for clarity. Looking at the name it might mean `getIJKValueFromFlatIndex` which is fair give the name `getIJK`, so how about `getValueAtIJK`